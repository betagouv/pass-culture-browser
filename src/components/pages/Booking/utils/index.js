export { default as flattenErrors } from './flattenErrors'
export { default as getCalendarProvider } from './getCalendarProvider'
export { default as parseHoursByStockId } from './parseHoursByStockId'
