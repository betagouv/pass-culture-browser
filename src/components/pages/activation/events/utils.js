import get from 'lodash.get'

import fp from '../../../../utils/functionnals'

export const filterActivationOffers = offers =>
  offers
    .filter(o => o.id && o)
    .filter(o => o.mediations && o.mediations.length >= 1 && o)
    .filter(
      o => o.event && get(o, 'event.offerType.value') === 'EventType.ACTIVATION'
    )

export const mapActivationOffersToSelectOptions = offers =>
  offers.map(o => {
    const { id } = o
    // construction de l'URL de redirection
    const { offerId, id: mediationId } = get(o, 'mediations.0')
    const url = `/decouverte/${offerId}/${mediationId}`
    // construction du label de l'option dans le select
    const venueCity = get(o, 'venue.city')
    const venueName = get(o, 'venue.name')
    const venueCodePostal = get(o, 'venue.postalCode')
    const label = `${venueCodePostal} ${venueCity} - ${venueName}`
    return { id, label, url }
  })

export const orderActivationOffersByLabel = offers =>
  offers.sort((a, b) => {
    if (a.label > b.label) return 1
    if (a.label < b.label) return -1
    return 0
  })

export const parseActivationOffers = offers =>
  fp.pipe(
    filterActivationOffers,
    mapActivationOffersToSelectOptions,
    orderActivationOffersByLabel
  )(offers)
