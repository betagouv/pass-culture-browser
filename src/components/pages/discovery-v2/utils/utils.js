import selectCurrentRecommendation from '../selectors/selectCurrentRecommendation'
import selectNextRecommendation from '../selectors/selectNextRecommendation'
import selectPreviousRecommendation from '../selectors/selectPreviousRecommendation'

export const MINIMUM_DELAY_BEFORE_RELOAD_RECOMMENDATION_3_HOURS = 3 * 60 * 60 * 1000
export const MINIMUM_DELAY_BEFORE_UPDATING_SEED_3_HOURS = 3 * 60 * 60 * 1000

export const isDiscoveryStartupUrl = match => {
  const { params, url } = match
  const { mediationId } = params

  if (!mediationId) {
    return true
  }

  const matchDiscoveryStartupPathRegex = /\/decouverte?$/
  const matches = url.match(matchDiscoveryStartupPathRegex)
  if (!matches) {
    return false
  }
  return matches.length > 0
}

export const checkIfShouldReloadRecommendationsBecauseOfLongTime = state => {
  const now = Date.now()
  const { lastRecommendationsRequestTimestamp } = state || 0
  if (!lastRecommendationsRequestTimestamp) return true
  return (
    now >= lastRecommendationsRequestTimestamp + MINIMUM_DELAY_BEFORE_RELOAD_RECOMMENDATION_3_HOURS
  )
}

const noop = () => {}
export const getRecommendationSelectorByCardPosition = position => {
  switch (position) {
    case 'current':
      return selectCurrentRecommendation
    case 'previous':
      return selectPreviousRecommendation
    case 'next':
      return selectNextRecommendation
    default:
      return noop
  }
}
