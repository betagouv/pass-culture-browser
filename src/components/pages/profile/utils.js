export const getAvailableBalanceByType = walletBalance => ({ actual, max }) => {
  const maximumAvailableByType = max - actual
  const actualAvailableByType = Math.min(walletBalance, maximumAvailableByType)
  return formatDecimals(actualAvailableByType)
}

const formatDecimals = number => {
  const numberWithTwoDecimals = number.toFixed(2)
  return numberWithTwoDecimals.replace('.00', '')
}

export default getAvailableBalanceByType
