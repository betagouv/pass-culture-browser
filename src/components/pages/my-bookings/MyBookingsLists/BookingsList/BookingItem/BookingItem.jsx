import moment from 'moment'
import PropTypes from 'prop-types'
import React from 'react'
import { capitalize } from 'react-final-form-utils'
import { Link } from 'react-router-dom'

import Icon from '../../../../../layout/Icon'
import Ribbon from '../../../../../layout/Ribbon'
import { getTimezone } from '../../../../../../utils/timezone'

export const stringify = date => timeZone =>
  capitalize(
    moment(date)
      .tz(timeZone)
      .format('dddd DD/MM/YYYY Ã  H:mm')
  )

const getDetailsUrl = (bookingId, location, match) => {
  const { pathname, search } = location
  const { params } = match
  const { details } = params
  if (details === 'details') {
    return `${pathname}${search}`
  }
  return `${pathname}/details/${bookingId}${search}`
}

const BookingItem = ({
  booking,
  location,
  match,
  offer,
  ribbon
}) => {
  const {
    id: bookingId,
    stock,
    token,
    thumbUrl
  } = booking
  const { beginningDatetime } = stock
  const { label, type } = ribbon || {}
  const { product, venue } = offer
  const { name: productName } = product
  const { departementCode } = venue
  const detailsUrl = getDetailsUrl(bookingId, location, match)
  const timeZone = getTimezone(departementCode)
  const stringifyDate = beginningDatetime && stringify(beginningDatetime)(timeZone)
  return (
    <li
      className="my-bookings-my-booking"
      data-token={token.toLowerCase()}
    >
      <Link
        className="teaser-link"
        to={detailsUrl}
      >
        <div className="teaser-thumb">
          {thumbUrl && (
            <img
              alt=""
              src={thumbUrl}
            />)}
        </div>
        <div className="teaser-wrapper">
          <div className="my-bookings-heading">
            <div className="teaser-title">{productName}</div>
            <div className="teaser-sub-title">{stringifyDate || 'Permanent'}</div>
          </div>
          <div className="my-bookings-token">{token}</div>
        </div>
        <div className="teaser-arrow">
          {ribbon && (
            <Ribbon
              label={label}
              type={type}
            />)}
          <Icon
            className="teaser-arrow-img"
            svg="ico-next-S"
          />
        </div>
      </Link>
    </li>
  )
}

BookingItem.defaultProps = {
  ribbon: null
}

BookingItem.propTypes = {
  booking: PropTypes.shape({
    id: PropTypes.string,
    stock: PropTypes.shape({
      beginningDatetime: PropTypes.string
    }),
    thumbUrl: PropTypes.string,
    token: PropTypes.string.isRequired,
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
    search: PropTypes.string.isRequired
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      details: PropTypes.string
    }).isRequired
  }).isRequired,
  offer: PropTypes.shape({
    product: PropTypes.shape({
      name: PropTypes.string,
    }).isRequired,
    venue: PropTypes.shape({
      departementCode: PropTypes.string
    }).isRequired
  }).isRequired,
  ribbon: PropTypes.shape({
    label: PropTypes.string,
    type: PropTypes.string
  }),
}

export default BookingItem
