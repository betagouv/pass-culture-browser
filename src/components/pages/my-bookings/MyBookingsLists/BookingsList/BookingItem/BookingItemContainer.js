import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import BookingItem from './BookingItem'
import selectRecommendationById from '../../../selectors/selectRecommendationById'

export const mapStateToProps = (state, ownProps) => {
  const { booking } = ownProps
  const { recommendationId } = booking
  const recommendation = selectRecommendationById(state, recommendationId)
  return {
    recommendation,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(BookingItem)
