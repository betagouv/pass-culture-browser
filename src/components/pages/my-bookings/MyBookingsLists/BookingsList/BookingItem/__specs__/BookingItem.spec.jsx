import { shallow } from 'enzyme'
import React from 'react'
import { Link } from 'react-router-dom'

import BookingItem from '../BookingItem'
import Icon from '../../../../../../layout/Icon/Icon'
import Ribbon from '../../../../../../layout/Ribbon/Ribbon'
import { DEFAULT_THUMB_URL } from '../../../../../../../utils/thumb'

describe('src | components | pages | my-bookings | MyBookingsList | BookingList | BookingItem', () => {
  let props

  beforeEach(() => {
    props = {
      booking: {
        id: 'AE',
        isCancelled: false,
        qrCode: 'data:image/png;base64,iVIVhzdjeizfjezfoizejojczez',
        stock: {},
        token: 'G9G9G9',
        thumbUrl: '/mediations/AE',
      },
      isQrCodeFeatureDisabled: true,
      location: {
        pathname: '/reservations',
        search: '?',
      },
      match: {
        params: {
          details: undefined,
        },
      },
      offer: {
        id: 'AE',
        name: 'Un livre pas mal',
        product: {
          name: 'Un livre pas mal',
          type: 'thing',
        },
        venue: {
          departementCode: '93',
        },
      },
      stock: { id: 'future stock 1', beginningDatetime: '2030-08-21T20:00:00Z' },
      trackConsultOffer: jest.fn(),
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a booking including token when qr code feature is inactive', () => {
    // given
    props.isQrCodeFeatureDisabled = true

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingLink = wrapper.find(Link).first()
    const bookingThumb = wrapper.find('.teaser-thumb').find('img')
    const bookingName = wrapper.find('.teaser-title-booking')
    const bookingDate = wrapper.find('.teaser-sub-title')
    const bookingToken = wrapper.find('.mb-token-flipped')
    const arrowIcon = wrapper.find('.teaser-arrow').find(Icon)
    const ribbon = wrapper.find('.teaser-arrow').find(Ribbon)

    expect(bookingLink).toHaveLength(1)
    expect(bookingLink.prop('to')).toBe('/reservations/details/AE?')
    expect(bookingThumb).toHaveLength(1)
    expect(bookingThumb.prop('alt')).toBe('')
    expect(bookingThumb.prop('src')).toBe('/mediations/AE')
    expect(bookingName.text()).toBe('Un livre pas mal')
    expect(bookingDate.text()).toBe('Mercredi 21/08/2030 à 22:00')
    expect(bookingToken).toHaveLength(1)
    expect(bookingToken.text()).toBe('g9g9g9')
    expect(ribbon).toHaveLength(0)
    expect(arrowIcon).toHaveLength(1)
  })

  it('should render a booking with no token when qr code feature is active', () => {
    // given
    props.isQrCodeFeatureDisabled = false

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingLink = wrapper.find(Link).first()
    const bookingThumb = wrapper.find('.teaser-thumb').find('img')
    const bookingName = wrapper.find('.teaser-title-booking')
    const bookingDate = wrapper.find('.teaser-sub-title')
    const bookingToken = wrapper.find('.mb-token-flipped')
    const arrowIcon = wrapper.find('.teaser-arrow').find(Icon)
    const ribbon = wrapper.find('.teaser-arrow').find(Ribbon)

    expect(bookingLink).toHaveLength(1)
    expect(bookingLink.prop('to')).toBe('/reservations/details/AE?')
    expect(bookingThumb).toHaveLength(1)
    expect(bookingThumb.prop('alt')).toBe('')
    expect(bookingThumb.prop('src')).toBe('/mediations/AE')
    expect(bookingName.text()).toBe('Un livre pas mal')
    expect(bookingDate.text()).toBe('Mercredi 21/08/2030 à 22:00')
    expect(bookingToken).toHaveLength(0)
    expect(ribbon).toHaveLength(0)
    expect(arrowIcon).toHaveLength(1)
    expect(arrowIcon.prop('svg')).toBe('ico-next-S')
  })

  it('should render a booking with a date when its an event', () => {
    // given
    props.stock.beginningDatetime = '2019-07-08T20:00:00Z'
    props.stock.endDatetime = '2019-07-08T23:00:00Z'

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingDate = wrapper.find('.teaser-sub-title').text()
    expect(bookingDate).toBe('Lundi 08/07/2019 à 22:00')
  })

  it('should render a booking with no date when its not an event', () => {
    // given
    props.stock.beginningDatetime = null
    props.stock.endDatetime = null

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingDate = wrapper.find('.teaser-sub-title').text()
    expect(bookingDate).toBe('Permanent')
  })

  it('should render a booking with a thumb url on thumb when mediation is provided', () => {
    // given
    props.booking.thumbUrl = '/mediations/AE'

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingThumb = wrapper.find('.teaser-thumb').find('img')
    expect(bookingThumb.prop('src')).toBe('/mediations/AE')
  })

  it('should render a booking with a default thumb url on thumb when no mediation is provided', () => {
    // given
    props.booking.thumbUrl = null

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingThumb = wrapper.find('.teaser-thumb').find('img')
    expect(bookingThumb.prop('src')).toBe(DEFAULT_THUMB_URL)
  })

  it('should render a booking with a Ribbon', () => {
    // given
    props.ribbon = {
      label: 'Annulé',
      type: 'cancelled',
    }

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const ribbon = wrapper.find(Ribbon)
    expect(ribbon).toHaveLength(1)
  })

  it('should render a link to booking qr code when offer is not used and qr code feature is active', () => {
    // given
    props.isQrCodeFeatureDisabled = false
    props.booking.qrCode = 'data:image/png;base64,iVIVhzdjeizfjezfoizejojczez'

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingTokenLink = wrapper.find('.mb-token-container').find(Link)
    expect(bookingTokenLink).toHaveLength(1)
    expect(bookingTokenLink.prop('to')).toBe('/reservations/details/AE?/qrcode')
  })

  it('should not render a link to booking qr code when offer is used and qr code feature is active', () => {
    // given
    props.isQrCodeFeatureDisabled = false
    props.booking.qrCode = null

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingTokenLink = wrapper.find('.mb-token-container').find(Link)
    expect(bookingTokenLink).toHaveLength(0)
  })

  it('should not render a link to booking qr code when offer is not used and qr code feature is inactive', () => {
    // given
    props.isQrCodeFeatureDisabled = true
    props.booking.qrCode = 'data:image/png;base64,iVIVhzdjeizfjezfoizejojczez'

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingTokenLink = wrapper.find('.mb-token-container').find(Link)
    expect(bookingTokenLink).toHaveLength(0)
  })

  it('should not render a link to booking qr code when offer is used and qr code feature is inactive', () => {
    // given
    props.isQrCodeFeatureDisabled = true
    props.booking.qrCode = null

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const bookingTokenLink = wrapper.find('.mb-token-container').find(Link)
    expect(bookingTokenLink).toHaveLength(0)
  })

  it('should render a token container when offer is not used, finished nor cancelled', () => {
    // given
    props.shouldDisplayToken = true

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const tokenContainer = wrapper.find('.mb-token-container')
    expect(tokenContainer).toHaveLength(1)
  })

  it('should not render a token container when offer is used, finished or cancelled', () => {
    // given
    props.shouldDisplayToken = false

    // when
    const wrapper = shallow(<BookingItem {...props} />)

    // then
    const tokenContainer = wrapper.find('.mb-token-container')
    expect(tokenContainer).toHaveLength(0)
  })
})
