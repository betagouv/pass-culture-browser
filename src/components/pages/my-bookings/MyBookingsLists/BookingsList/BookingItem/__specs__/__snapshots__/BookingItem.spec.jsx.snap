// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | my-bookings | MyBookingsList | BookingList | BookingItem should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookingItem
    booking={
      Object {
        "id": "AE",
        "isCancelled": false,
        "qrCode": "data:image/png;base64,iVIVhzdjeizfjezfoizejojczez",
        "stock": Object {},
        "thumbUrl": "/mediations/AE",
        "token": "G9G9G9",
      }
    }
    isQrCodeFeatureDisabled={true}
    location={
      Object {
        "pathname": "/reservations",
        "search": "?",
      }
    }
    match={
      Object {
        "params": Object {
          "details": undefined,
        },
      }
    }
    offer={
      Object {
        "id": "AE",
        "name": "Un livre pas mal",
        "product": Object {
          "name": "Un livre pas mal",
          "type": "thing",
        },
        "venue": Object {
          "departementCode": "93",
        },
      }
    }
    ribbon={null}
    shouldDisplayToken={true}
    stock={
      Object {
        "beginningDatetime": "2030-08-21T20:00:00Z",
        "id": "future stock 1",
      }
    }
    trackConsultOffer={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef
          className="teaser-link"
          onClick={[MockFunction]}
          to="/reservations/details/AE?"
        >
          <div
            className="teaser-thumb"
          >
            <img
              alt=""
              src="/mediations/AE"
            />
          </div>
          <div
            className="teaser-wrapper"
          >
            <div
              className="mb-corner bottom left"
            />
            <div
              className="mb-corner bottom right"
            />
            <div
              className="mb-heading"
            >
              <div
                className="teaser-title-booking"
              >
                Un livre pas mal
              </div>
              <div
                className="teaser-sub-title"
              >
                Mercredi 21/08/2030 à 22:00
              </div>
            </div>
          </div>
          <div
            className="teaser-arrow"
          >
            <Icon
              alt={null}
              className="teaser-arrow-img"
              src={null}
              svg="ico-next-S"
            />
          </div>
        </ForwardRef>,
        <div
          className="mb-token-container"
        >
          <div
            className="mb-corner top left"
          />
          <div
            className="mb-corner top right"
          />
          <div
            className="mb-token-title"
          >
            Votre contremarque
          </div>
          <div
            className="mb-token-contremarque-container"
          >
            <div
              className="mb-token-flipped"
            >
              g9g9g9
            </div>
          </div>
        </div>,
      ],
      "className": "mb-my-booking",
      "data-token": "g9g9g9",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="teaser-thumb"
            >
              <img
                alt=""
                src="/mediations/AE"
              />
            </div>,
            <div
              className="teaser-wrapper"
            >
              <div
                className="mb-corner bottom left"
              />
              <div
                className="mb-corner bottom right"
              />
              <div
                className="mb-heading"
              >
                <div
                  className="teaser-title-booking"
                >
                  Un livre pas mal
                </div>
                <div
                  className="teaser-sub-title"
                >
                  Mercredi 21/08/2030 à 22:00
                </div>
              </div>
            </div>,
            <div
              className="teaser-arrow"
            >
              <Icon
                alt={null}
                className="teaser-arrow-img"
                src={null}
                svg="ico-next-S"
              />
            </div>,
            null,
          ],
          "className": "teaser-link",
          "onClick": [MockFunction],
          "to": "/reservations/details/AE?",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                src="/mediations/AE"
              />,
              "className": "teaser-thumb",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "/mediations/AE",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="mb-corner bottom left"
                />,
                <div
                  className="mb-corner bottom right"
                />,
                <div
                  className="mb-heading"
                >
                  <div
                    className="teaser-title-booking"
                  >
                    Un livre pas mal
                  </div>
                  <div
                    className="teaser-sub-title"
                  >
                    Mercredi 21/08/2030 à 22:00
                  </div>
                </div>,
              ],
              "className": "teaser-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "mb-corner bottom left",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "mb-corner bottom right",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="teaser-title-booking"
                    >
                      Un livre pas mal
                    </div>,
                    <div
                      className="teaser-sub-title"
                    >
                      Mercredi 21/08/2030 à 22:00
                    </div>,
                  ],
                  "className": "mb-heading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Un livre pas mal",
                      "className": "teaser-title-booking",
                    },
                    "ref": null,
                    "rendered": "Un livre pas mal",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        false,
                        "Mercredi 21/08/2030 à 22:00",
                      ],
                      "className": "teaser-sub-title",
                    },
                    "ref": null,
                    "rendered": Array [
                      false,
                      "Mercredi 21/08/2030 à 22:00",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                alt={null}
                className="teaser-arrow-img"
                src={null}
                svg="ico-next-S"
              />,
              "className": "teaser-arrow",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": null,
                "className": "teaser-arrow-img",
                "src": null,
                "svg": "ico-next-S",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          null,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mb-corner top left"
            />,
            <div
              className="mb-corner top right"
            />,
            <div
              className="mb-token-title"
            >
              Votre contremarque
            </div>,
            <div
              className="mb-token-contremarque-container"
            >
              <div
                className="mb-token-flipped"
              >
                g9g9g9
              </div>
            </div>,
            false,
          ],
          "className": "mb-token-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mb-corner top left",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mb-corner top right",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Votre contremarque",
              "className": "mb-token-title",
            },
            "ref": null,
            "rendered": "Votre contremarque",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="mb-token-flipped"
              >
                g9g9g9
              </div>,
              "className": "mb-token-contremarque-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "g9g9g9",
                "className": "mb-token-flipped",
              },
              "ref": null,
              "rendered": "g9g9g9",
              "type": "div",
            },
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef
            className="teaser-link"
            onClick={[MockFunction]}
            to="/reservations/details/AE?"
          >
            <div
              className="teaser-thumb"
            >
              <img
                alt=""
                src="/mediations/AE"
              />
            </div>
            <div
              className="teaser-wrapper"
            >
              <div
                className="mb-corner bottom left"
              />
              <div
                className="mb-corner bottom right"
              />
              <div
                className="mb-heading"
              >
                <div
                  className="teaser-title-booking"
                >
                  Un livre pas mal
                </div>
                <div
                  className="teaser-sub-title"
                >
                  Mercredi 21/08/2030 à 22:00
                </div>
              </div>
            </div>
            <div
              className="teaser-arrow"
            >
              <Icon
                alt={null}
                className="teaser-arrow-img"
                src={null}
                svg="ico-next-S"
              />
            </div>
          </ForwardRef>,
          <div
            className="mb-token-container"
          >
            <div
              className="mb-corner top left"
            />
            <div
              className="mb-corner top right"
            />
            <div
              className="mb-token-title"
            >
              Votre contremarque
            </div>
            <div
              className="mb-token-contremarque-container"
            >
              <div
                className="mb-token-flipped"
              >
                g9g9g9
              </div>
            </div>
          </div>,
        ],
        "className": "mb-my-booking",
        "data-token": "g9g9g9",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="teaser-thumb"
              >
                <img
                  alt=""
                  src="/mediations/AE"
                />
              </div>,
              <div
                className="teaser-wrapper"
              >
                <div
                  className="mb-corner bottom left"
                />
                <div
                  className="mb-corner bottom right"
                />
                <div
                  className="mb-heading"
                >
                  <div
                    className="teaser-title-booking"
                  >
                    Un livre pas mal
                  </div>
                  <div
                    className="teaser-sub-title"
                  >
                    Mercredi 21/08/2030 à 22:00
                  </div>
                </div>
              </div>,
              <div
                className="teaser-arrow"
              >
                <Icon
                  alt={null}
                  className="teaser-arrow-img"
                  src={null}
                  svg="ico-next-S"
                />
              </div>,
              null,
            ],
            "className": "teaser-link",
            "onClick": [MockFunction],
            "to": "/reservations/details/AE?",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  src="/mediations/AE"
                />,
                "className": "teaser-thumb",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "/mediations/AE",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="mb-corner bottom left"
                  />,
                  <div
                    className="mb-corner bottom right"
                  />,
                  <div
                    className="mb-heading"
                  >
                    <div
                      className="teaser-title-booking"
                    >
                      Un livre pas mal
                    </div>
                    <div
                      className="teaser-sub-title"
                    >
                      Mercredi 21/08/2030 à 22:00
                    </div>
                  </div>,
                ],
                "className": "teaser-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "mb-corner bottom left",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "mb-corner bottom right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="teaser-title-booking"
                      >
                        Un livre pas mal
                      </div>,
                      <div
                        className="teaser-sub-title"
                      >
                        Mercredi 21/08/2030 à 22:00
                      </div>,
                    ],
                    "className": "mb-heading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Un livre pas mal",
                        "className": "teaser-title-booking",
                      },
                      "ref": null,
                      "rendered": "Un livre pas mal",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          false,
                          "Mercredi 21/08/2030 à 22:00",
                        ],
                        "className": "teaser-sub-title",
                      },
                      "ref": null,
                      "rendered": Array [
                        false,
                        "Mercredi 21/08/2030 à 22:00",
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  alt={null}
                  className="teaser-arrow-img"
                  src={null}
                  svg="ico-next-S"
                />,
                "className": "teaser-arrow",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alt": null,
                  "className": "teaser-arrow-img",
                  "src": null,
                  "svg": "ico-next-S",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            null,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mb-corner top left"
              />,
              <div
                className="mb-corner top right"
              />,
              <div
                className="mb-token-title"
              >
                Votre contremarque
              </div>,
              <div
                className="mb-token-contremarque-container"
              >
                <div
                  className="mb-token-flipped"
                >
                  g9g9g9
                </div>
              </div>,
              false,
            ],
            "className": "mb-token-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "mb-corner top left",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "mb-corner top right",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Votre contremarque",
                "className": "mb-token-title",
              },
              "ref": null,
              "rendered": "Votre contremarque",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="mb-token-flipped"
                >
                  g9g9g9
                </div>,
                "className": "mb-token-contremarque-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "g9g9g9",
                  "className": "mb-token-flipped",
                },
                "ref": null,
                "rendered": "g9g9g9",
                "type": "div",
              },
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
